modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -87,30 +91,37 @@ public int getSize() {
    * @return
    */
   private <R> R performWithConnection(Function<Object, R> function) {
-    ConnectionSupplier connectionSupplier = getConnectionSupplier();
-    Object connection = getConnection(connectionSupplier);
+    actualConnectionSupplier = getActualConnectionSupplier();
+    Object connection = getConnection(actualConnectionSupplier);
     try {
       R result = function.apply(connection);
-      connectionSupplier.close();
+      actualConnectionSupplier.close();
       return result;
-    } catch (Exception e) {
+    } catch (Exception exception) {
       if (isFirstPage) {
-        closeDelegate(connection);
+        safely(() -> delegate.close(connection), e -> LOGGER.debug("Found exception closing buffer", e));
       }
-      extractConnectionException(e).ifPresent(ex -> connectionSupplier.invalidateConnection());
-      throw new MuleRuntimeException(createStaticMessage(COULD_NOT_EXECUTE), e);
+      extractConnectionException(exception).ifPresent(ex -> actualConnectionSupplier.invalidateConnection());
+      throw exception;
     }
   }
 
   /**
    * Closes the delegate
    */
   @Override
-  public void close() throws IOException {
-    ConnectionSupplier connectionSupplier = getConnectionSupplier();
-    closeDelegate(getConnection(connectionSupplier));
-    connectionSupplier.close();
-    connectionSupplierFactory.dispose();
+  public void close() {
+    try {
+      if (actualConnectionSupplier == null) {
```

Comment:
if there's no actual connection supplier then do nothing. You're opening a connection for the sole purpose of closing it